lib/notifier.js

```diff
@@ -263,7 +325,7 @@ function addListener(server, user, regex, event, callback) {
 
     // If the old API is used, convert all arguments to the new API
     // If `event` is a Worker, only 4 arguments are provided
-    if (arguments.length === 4 || arguments.length === 5) {
+    if (arguments.length > 3) {
```

Comment:
Why not use the original - and more precise - condition?

Reply:
this is used mainly to detect old/new function signature, if the caller passes more arguments then they will be ignored.